h2o-core/src/main/java/water/util/MRUtils.java

```diff
@@ -36,13 +34,19 @@ public static Frame sampleFrame(Frame fr, final long rows, final long seed) {
       @Override
       public void map(Chunk[] cs, NewChunk[] ncs) {
         final Random rng = getRNG(0);
+        final BufferedString bStr = new BufferedString();
         int count = 0;
         for (int r = 0; r < cs[0]._len; r++) {
           rng.setSeed(seed+r+cs[0].start());
           if (rng.nextFloat() < fraction || (count == 0 && r == cs[0]._len-1) ) {
             count++;
             for (int i = 0; i < ncs.length; i++) {
-              ncs[i].addNum(cs[i].atd(r));
+              if (cs[i] instanceof CStrChunk)
```

Comment:
Just technical note: the code depends on implementation details (`instanceof C16Chunk`) - if we represent UUID with another type of chunk, then this will fail.

Better is to use vector types directly (however, then we pay a network cost to transfer them around)